lib/ansible/modules/remote_management/manageiq/manageiq_alert_profiles.py

```diff
@@ -0,0 +1,348 @@
+      - The resource type for the alert profile in ManageIQ. Required when state is "present".
+    choices: ['Vm', 'ContainerNode', 'MiqServer', 'Host', 'Storage', 'EmsCluster',
+              'ExtManagementSystem', 'MiddlewareServer']
+  alert_policies:
+    description:
+      - List of alert policy descriptions to assign to this profile.
+      - Required if state is "present"
+  notes:
+    description:
+      - Optional notes for this profile
+
+'''
+
+EXAMPLES = '''
+- name: List alert policies in ManageIQ
+  manageiq_alert_policies:
+    state: list
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Add an alert profile to ManageIQ
+  manageiq_alert_profiles:
+    state: present
+    description: Test profile
+    resource_type: ContainerNode
+    alert_policies:
+      - Test Alert 01
+      - Test Alert 02
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete an alert profile from ManageIQ
+  manageiq_alert_profiles:
+    state: absent
+    description: Test profile
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+# TODO: I copied this from yaacov's ansible/ansible PR 31233. It should be removed
+# from here once that PR is merged.
+def find_collection_resource_or_fail(module, manageiq, collection_name, **params):
+    """ Searches the collection resource by the collection name and the param passed.
+
+    Returns:
+        the resource as an object if it exists in manageiq, Fail otherwise.
+    """
+    resource = manageiq.find_collection_resource_by(collection_name, **params)
+    if resource:
+        return resource
+    else:
+        msg = "{collection_name} where {params} does not exist in manageiq".format(
+            collection_name=collection_name, params=str(params))
+    module.fail_json(msg=msg)
+
+
+class ManageIQAlertProfiles(object):
+    """ Object to execute alert profile management operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+        self.url = '{api_url}/alert_definition_profiles'.format(api_url=self.api_url)
+
+    def get_profiles(self):
+        """ Get all alert profiles from ManageIQ
+        """
+        try:
+            response = self.client.get(self.url + '?expand=alert_definitions,resources')
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query alert profiles: {error}".format(error=e))
+        return response.get('resources', [])
```

Comment:
If you like to replace `None` with `[]`, it sugguest to change it to `response.get('resources') or [] `. The difference is, currently with this code, `[]` will only returned if there is no `resources` key in `response`